### DISCUSSION TITLE
Map markers vs. My Places

### DISCUSSION THREAD

**Discussion by sonora** [osmandapp/OsmAnd#20115] [Ideas]
Users,ask me why Map markers are not a tab on the My Places screen, instead have their own screen and menu entry?

**Comment by vshcherb** [comment]
As a concept Markers should have been a tool to display Widget direction and create an itinerary list. Instead they combined an entity which also stores anonymous waypoints. That was a mistake.
Though we think and want to split "Direction-tool" and "Anonymous waypoint" as 2 separate entities than waypoints will be long to My Places and Map Markers -> "Directions". And any point could become a point to be used in direction tool / itinerary.

This change is quite big and creates 1-more entity which could be even more confusing so Map Markers still have own list and own widget.

**Comment by sonora** [comment]
Rather than creating even mote complexity, I vote for less:
- Let's make our current 'Favorites' a special case of 'Tracks'. Each Favorite category would be a track with just waypoints (in a respective points group). We already have this today, even if still in a dedicated folder.
- Let's make Map markers (if needed also intinerary waypoints) a temporary case of 'Favorites', in each in a dedicated GPX file.
-  Both are accessible via the Tracks mechanism, as tracks. But also in addition, as needed, there may be dedicated screens (tabs on MyPlaces or the itineraries), to make them accessible also more directly frim where needed.
- But basically all share the same code base 
- And are stored in the same fashion. Only access for creation/deletion can be shortcut, as needed.

**Reply by vshcherb** [reply]
Favorites are already Tracks, we store favorites internally each group as GPX file, so we don't need to change anything and we don't want to change the UI concept yet.

We can't change Markers without changing functionality they also stored in GPX as backup but Markers are not indpendent they can be linked with favorite group or track waypoints or be separate. It's anyway 2 separate enitities
1. Entity that stores points could be favorite group (though they have special icon as a Flag)
2. Entity that used to be displayed in widget and the itinerary list (to keep passed and non-passed poitns)

**Comment by ZeljkoBG** [comment]
Markers should be temporary points that are deleted automatically and Favorites are recorded

**Reply by pebogufi** [reply]
> Markers should be temporary points that are deleted automatically and Favorites are recorded

No no no !
You never know, how people use it.
